kasa/smart/modules/waterleaksensor.py

```diff
@@ -62,3 +75,14 @@ def status(self) -> WaterleakStatus:
     def alert(self) -> bool:
         """Return true if alarm is active."""
         return self._device.sys_info["in_alarm"]
+
+    @property
+    def alert_timestamp(self) -> datetime | None:
+        """Return timestamp of the last leak trigger."""
+        # The key is not always be there, maybe if it hasn't ever been triggered?
+        if "trigger_timestamp" not in self._device.sys_info:
+            return None
+
+        ts = self._device.sys_info["trigger_timestamp"]
+        tz = self._device.modules[Module.Time].timezone
+        return datetime.fromtimestamp(ts, tz=tz)
```

Comment:
Why isn't this getting picked up by tests?

Reply:
There's no fixture with the data available. I'll add one so that the both paths get covered.

Reply:
Added a separate `_with_trigger_timestamp.json`.

Reply:
As that's now breaking the `generate_supported` I'd suggest just adding the timestamp to the main one and not worrying about the `None` path.

Reply:
Agreed, that's the nicest solution here & the behavior is commented in the code so it's fine.